dma/grdmac2/grdmac2.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Phoenix-RTOS
+ *
+ * GRLIB DMA Controller driver
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include <sys/mman.h>
+#include <sys/platform.h>
+
+#if defined(__CPU_GR716)
+#include <phoenix/arch/sparcv8leon3/gr716/gr716.h>
+#elif defined(__CPU_GR712RC)
+#include <phoenix/arch/sparcv8leon3/gr712rc/gr712rc.h>
+#else
+#error "Unsupported target"
+#endif
```

Comment:
Would be nice to move this to some common header file, as it's needed in multiple .c files in this driver